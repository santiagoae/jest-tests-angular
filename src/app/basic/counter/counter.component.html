<main class="container">
    <h1>Counter {{ counter }}</h1>
    <hr>
    <button class="btn btn-dark me-2" (click)="increaseBy(1)">+1</button>
    <button class="btn btn-dark" (click)="increaseBy(-1)">-1</button>
</main>
