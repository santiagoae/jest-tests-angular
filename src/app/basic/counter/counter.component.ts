import { Component } from '@angular/core';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'app-counter',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './counter.component.html',
  styleUrl: './counter.component.css'
})
export class CounterComponent {
  counter:number = 10;

  increaseBy(value:number){
    this.counter += value;
    // TODO::
    console.log({newValue: this.counter}); 
  }
}
