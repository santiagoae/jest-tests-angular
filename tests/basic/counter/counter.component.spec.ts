import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from '../../../src/app/basic/counter/counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [CounterComponent]
    })
    .compileComponents();
    
    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should be match with the snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  test('increaseBy should be increase based on the argument', () => {
    component.increaseBy(5);
    expect(component.counter).toBe(15);
  });

  test('click on buttons should be increment and decrease in 1', () => {
    const buttons = compiled.querySelectorAll('button');//se puede buscar por clases . por id # por algo especifico []
    buttons[0].click();
    expect(component.counter).toBe(11);

    buttons[1].click();
    buttons[1].click();
    expect(component.counter).toBe(9);
  });

  test('changing counter should be update h1 label', () => {
    component.increaseBy(10);
    fixture.detectChanges();
    
    const h1 = compiled.querySelector('h1');
    expect(h1?.textContent).toContain('20');//que en su estring tenga un valor de 20
  });
});
