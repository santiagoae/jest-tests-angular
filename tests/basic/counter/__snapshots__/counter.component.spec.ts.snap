// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent match with the snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Counter 10
  </h1>
  <hr />
  <button>
    +1
  </button>
  <button>
    -1
  </button>
</div>
`;

exports[`CounterComponent should be match with the snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Counter 10
  </h1>
  <hr />
  <button>
    +1
  </button>
  <button>
    -1
  </button>
</div>
`;
